src/pool-bin/BinPoolManager.sol

```diff
@@ -175,6 +175,55 @@ contract BinPoolManager is IBinPoolManager, ProtocolFees, Extsload {
         vault.accountAppBalanceDelta(key, delta, msg.sender);
     }
 
+    function swapExactOut(PoolKey memory key, bool swapForY, uint128 amountOut, bytes calldata hookData)
+        external
+        whenNotPaused
+        returns (BalanceDelta delta)
+    {
+        if (amountOut == 0) revert InsufficientAmountIn();
+
+        PoolId id = key.toId();
+        BinPool.State storage pool = pools[id];
+        pool.checkPoolInitialized();
+
+        (uint128 amountToSwap, BeforeSwapDelta beforeSwapDelta, uint24 lpFeeOverride) =
+            BinHooks.beforeSwap(key, swapForY, amountOut, hookData);
```

Comment:
Is it possible we merge this with the original `swap()`,  otherwise we might need another `BinHooks.beforeSwap` here 

what about considering change uint128 amount to int128 amount
And then positive value denotes user specifies output ..

Reply:
agree, we should merge and have the same behaviour of how cl pool behave -- eg. negative input = exactInput and exact input = exactOutput